common/common.cpp

```diff
@@ -0,0 +1,43 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#include <common/common.hpp>
+#include <common/error.hpp>
+
+#include <cerrno>
+#include <cstdlib>
+
+namespace mender {
+namespace common {
+
+mender::common::expected::ExpectedLong StringToLong(const string &str, int base) {
```

Comment:
Should the whole function be `noexcept` then also? :thinking:

Reply:
> There are `std::stol()` and `std::stoll()`: https://en.cppreference.com/w/cpp/string/basic_string/stol. I take it we can't use any of them because they rely on exceptions, so I'd add a comment here saying so.

Yes, exactly!

> Should the whole function be `noexcept` then also? thinking

In principle yes, but all our functions are `noexcept`, so instead of spreading this around everywhere I propose to get the `-fno-exception` change that @vpodzime suggested in, and then it will automatically cover all of them.